### PR TITLE
Ensure binary rules are made executable by rex backend

### PR DESCRIPTION
**Pull request by henryaj**
Fixes problems we were seeing with .pex files not being executable.

Is there a good reason we weren't doing this already?

### CODE REVIEW

File: src/remote/action.go [hunk 1]
```diff
@@ -135,7 +135,11 @@ func (c *Client) getCommand(target *core.BuildTarget) string {
 		}
 		return cmd
 	}
-	return target.GetCommand(c.state)
+	cmd := target.GetCommand(c.state)
+	if target.IsBinary {
+		return "(" + cmd + ") && chmod +x $OUT"
```
Comment (peterebden) [inline src/remote/action.go]: might need to be $OUTS (don't think we necessarily guarantee binary targets have a single output, although it is a good idea for them to)

**Comment by peterebden** [comment]
> Is there a good reason we weren't doing this already?

I made plz enforce it itself at some point early on (because it was annoying to keep forgetting it and easy for it to do). Unfortunate we can't do that here :(

**Comment by henryaj** [comment]
This is causing problems with `sh_cmd` targets (which use a heredoc, so can't be appended to); will need to rethink a bit

**Comment by henryaj** [comment]
Ready for review.

**Pull request merged**